aws/fedora-coreos/kubernetes/ami.tf

```diff
@@ -13,16 +13,8 @@ data "aws_ami" "fedora-coreos" {
     values = ["hvm"]
   }
 
-  filter {
-    name   = "name"
-    values = ["fedora-coreos-31.*.*.*-hvm"]
-  }
-
   filter {
     name   = "description"
-    values = ["Fedora CoreOS stable*"]
+    values = ["Fedora CoreOS ${var.os_stream} *"]
   }
-
-  # try to filter out dev images (AWS filters can't)
-  name_regex = "^fedora-coreos-31.[0-9]*.[0-9]*.[0-9]*-hvm*"
```

Comment:
Originally this filtered out development images published in us-east-1. Checked this and with the new description filter, `next` stream correctly picks the image listed on https://getfedora.org/en/coreos/download?tab=cloud_launchable&stream=next, not next-devel (i.e. development)